### ISSUE TITLE
enable vhosts

### ISSUE THREAD

**Issue by J77J** [dokku/dokku#2885]
I was under the impression (wrongly it appears) that setting up dokku on centos would mean that virtual hosts would be enabled by default..

Anyway - I'm now trying to do this after the fact..

How would I do that? I've found some instructions that say to alter the VHOST file in the dokku directory - however, there isn't one. There is a HOSTNAME file, is this the file to edit?

from here: http://dokku.viewdocs.io/dokku/configuration/domains/#customizing-hostnames

> The subdomain is inferred from the pushed application name, while the domain.tld is set during initial configuration and stored in the $DOKKU_ROOT/VHOST file. It can then be modified with dokku domains:add-global

so would I just have to set it with dokku domains:add-global and that would then enable virtual hosts?
also:

> Adding a domain before deploying an application will result in port mappings being set.

so - I should not add subdomains at all and just let dokku create them?

also - do i then need to delete and redeploy the app I have deployed? Or is there a way to alter what is there now to save that effort?

Basically - I want it set up so that any subsequent deploys of apps are done with subdomains and not ports - any pointers? or if someone could point me to some instructions that show this I'd be grateful..

Johnny

**Comment by josegonzalez** [comment]
@J77J did you figure this out?

**Comment by J77J** [comment]
Partially, I managed to get a VHOSTS file created, so now when I try to deploy it looks like that is working, however the database does not seem to want to connect.

```
       PG::ConnectionBad: could not connect to server: Connection refused
        Is the server running on host "localhost" (127.0.0.1) and accepting
        TCP/IP connections on port 5432?
       could not connect to server: Network is unreachable
        Is the server running on host "localhost" (::1) and accepting
        TCP/IP connections on port 5432?
```
I'm not sure if I have to unlink and then relink due to me having the app there before? Or is this unrelated to dokku?

**Comment by josegonzalez** [comment]
Thats a separate issue. Please ensure that you have both created the service and linked it to your application. You can get that info from:

```
dokku postgres:info SERVICE
```

**Comment by J77J** [comment]
Yes - just had to link..

Thanks Jose!
